modules/gapi/test/render/ftp_render_test.cpp

```diff
@@ -0,0 +1,64 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+
+#ifdef HAVE_FREETYPE
+
+#include <random>
+
+#include "../test_precomp.hpp"
+#include "api/ftp_render.hpp"
+
+namespace opencv_test
+{
+    inline void RunTest(const std::string& font,
+                        size_t num_iters,
+                        size_t lower_char_code,
+                        size_t upper_char_code)
+    {
+        cv::gapi::wip::draw::FTPTextRender ftpr(font);
+
+        std::mt19937 gen{std::random_device()()};
+        std::uniform_int_distribution<int> dist(lower_char_code, upper_char_code);
+        std::uniform_int_distribution<int> dist_size(2, 200);
+
+        for (size_t i = 0; i < num_iters; ++i)
+        {
+            size_t text_size = dist_size(gen);
+            std::wstring text;
+
+            for (size_t j = 0; j < text_size; ++j)
+            {
+                wchar_t c = dist(gen);
+                text += c;
+            }
+
+            int fh       = dist_size(gen);
+            int baseline = 0;
+            cv::Size size;
+
+            ASSERT_NO_THROW(size = ftpr.getTextSize(text, fh, &baseline));
+
+            cv::Mat bmp(size, CV_8UC1, cv::Scalar::all(0));
+            cv::Point org(0, bmp.rows - baseline);
+
+            ASSERT_NO_THROW(ftpr.putText(bmp, text, org, fh));
+        }
+    }
+
+    TEST(FTPTextRender, Smoke_Test_Ascii)
+    {
+        RunTest("/usr/share/fonts/truetype/wqy/wqy-microhei.ttc", 2000, 32, 126);
```

Comment:
These files may be missing on the target system. What if it runs on Windows?

The correct way to do this is to add the requried files to `opencv_extra` and then to refer to those via `findDataFile()`, I believe (cc: @alalek )

Reply:
Fixed